app/src/main/java/com/example/demomvp/ui/login/LoginPresenter.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.demomvp.ui.login
+
+import android.text.TextUtils
+import com.example.demomvp.MainApp
+import com.example.demomvp.R
+import com.example.demomvp.data.model.User
+import com.example.demomvp.data.repository.LoginRepository
+
+class LoginPresenter(
+    private val view: LoginContact.View,
+    private val repository: LoginRepository
+) : LoginContact.Presenter {
+    override fun onClickLogin(userName: String, pass: String) {
+        if (isValidate(userName, pass)) {
+            MainApp.getPrefs().setIsLogin(true)
+            view.loginSuccess(User(userName, pass))
+        }
+    }
+
+    override fun onClickFakeRegister() {
+        repository.addUser(User("a", "b"))
+    }
```

Comment:
Đặt tên hàm theo nội dung mà hàm sẽ thực hiện 
```suggestion
    override fun addUser(user: User) {
        repository.addUser(User("a", "b"))
    }
```